Formula/amp.rb

```diff
@@ -0,0 +1,44 @@
+class Amp < Formula
+  desc "Text editor for your terminal"
+  homepage "https://amp.rs"
+  url "https://github.com/jmacdonald/amp/archive/0.3.2.tar.gz"
+  sha256 "9080c894111149d58a0b805e0c08edaf1d9e7cd56c4b955ba9111feec6b55208"
+  head "https://github.com/jmacdonald/amp.git"
+
+  depends_on "rust" => :build
+  depends_on "cmake" => :build
+  depends_on "openssl"
```

Comment:
This seems to get ignored in some cases. I have both `openssl@1.1` and `openssl` installed locally and it has chosen the former:

```
/usr/local/bin/amp:
	/System/Library/Frameworks/AppKit.framework/Versions/C/AppKit (compatibility version 45.0.0, current version 1561.40.104)
	/System/Library/Frameworks/Foundation.framework/Versions/C/Foundation (compatibility version 300.0.0, current version 1452.19.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.50.4)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 1.0.0, current version 58286.50.200)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1452.19.0)
	/usr/lib/libcurl.4.dylib (compatibility version 7.0.0, current version 9.0.0)
	/usr/local/opt/openssl@1.1/lib/libssl.1.1.dylib (compatibility version 1.1.0, current version 1.1.0)
	/usr/local/opt/openssl@1.1/lib/libcrypto.1.1.dylib (compatibility version 1.1.0, current version 1.1.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/usr/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
```

Reply:
Yep, the workaround I've been giving folks who have failed to build locally using the 3rd-party tap is to temporarily uninstall `openssl@1.1`, which has always done the trick. I'm hoping that a pre-built bottle, compiled in a CI environment where `openssl@1.1` isn't installed will solve that.

Reply:
It must link the correct OpenSSL even if both are installed regardless of bottles.

Reply:
`tectonic` sets
```
    ENV["OPENSSL_INCLUDE_DIR"] = Formula["openssl"].opt_include
    ENV["DEP_OPENSSL_INCLUDE"] = Formula["openssl"].opt_include
    ENV["OPENSSL_LIB_DIR"] = Formula["openssl"].opt_lib
```
I don't know if those are relevant here?

Reply:
Does Homebrew link bottles after they're downloaded? I thought that, after being compiled and linked once in CI, the rest was up to dynamic linking when the executable is loaded/run by the kernel.

Reply:
Those environment flags are worth a shot; incoming. 🙂

Reply:
@jmacdonald even though we have bottles, users can still do local source builds. When they do local source builds, it needs to link the correct OpenSSL. So the bottles aren't relevant.

Reply:
@DomT4 mind checking whether any/all of those envvars do the trick for you?

Reply:
Not sure if all three are necessary yet, but in combination they do work:
```
	/usr/local/opt/openssl/lib/libssl.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/local/opt/openssl/lib/libcrypto.1.0.0.dylib (compatibility version 1.0.0, current version 1.0.0)
```

Reply:
Nice! That fixes the source build for folks with `openssl@1.1`, but I've added those flags to the build script and it's still failing in CI.

I think the openssl issue might be a red herring, there. The initial formula's test process launched `amp`, sent it a `SIGINT` signal, and collected the process status, which was zero. If there was a missing dependency, I'm pretty sure that value would be non-zero. The trick is, there's no way to see what specifically is failing as part of the `expect` portion of the test.

Reply:
@jmacdonald yeah @DomT4 wasn't suggesting a connection between the two problems ;)

Reply:
`ENV["DEP_OPENSSL_INCLUDE"] = Formula["openssl"].opt_include` seemingly can be dropped. It works fine with just the other two.